### PR TITLE
Add tests to Travis CI

### PR DESCRIPTION
**Pull request by kchitalia**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,5 +15,14 @@ env:
       # install timeout in minutes
       - ADB_INSTALL_TIMEOUT=8
 
+# Create and start the emulator
+before_script:
+   - echo no | android create avd --force -n test -t android-21  --abi armeabi-v7a -c 128M
```
Comment (bhariharan) [inline .travis.yml]: I think we should change the ```abi``` to ```x86```. ```arm``` is too slow. Is ```-c``` for SD card?

Comment (kchitalia) [inline .travis.yml]: Yep -c is for SD card. Will change to x86

**Comment by wmathurin** [comment]
Looks fine. Should we run the tests against different emulators (different OS versions - may be min and max that we support)? 
Should do the same on iOS (build/test against min and max OS versions we support)?

File: .travis.yml (hunk 1 above)
Comment (kchitalia) [inline .travis.yml]: Apparently there are some issues with the x86 emulator:

https://github.com/travis-ci/travis-ci/issues/1419

But based on some suggestions on that thread, I could use the -PdisablePreDex to speed up the build.

**Comment by bhariharan** [comment]
Matrix builds are ideal, but we faced tonnes of issues with the variants in the past with ```Jenkins```. I hope ```Travis``` works better :-)

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**